### PR TITLE
utils/memory : use block string instead of block id

### PR DESCRIPTION
**Pull request by abdhaleegit**
It is easier for the user to pass full string of block id
like 'memory234' instead of just '234', it saves few steps of the
user to find the memory blocks and call this functions to hotplug/unplug

fix: in _check_memory_state() no try for 120 times to check for state. So
2 retries is enough for the state to change from 'going-online' to 'online'

Signed-off-by: Abdul Haleem <abdhalee@linux.vnet.ibm.com>

### CODE REVIEW

File: avocado/utils/memory.py [hunk 1]
```diff
@@ -41,17 +41,17 @@ def _check_memory_state(block):
     Check the given memory block is online or offline
 
     :param block: memory block id.
-    :type string: like 198
+    :type string: like memory198
```
Comment (ldoktor) [inline avocado/utils/memory.py]: Please don't do incompatible changes like this as it breaks existing tests. I agree such usage is perhaps more natural and easier to use, so how about to accept both?

```python
if not block.startswith("memory"):
    block = "memory%s" % block
```

File: avocado/utils/memory.py [hunk 2]
```diff
@@ -41,17 +41,17 @@ def _check_memory_state(block):
...
     """
     def _is_online():
-        with open('/sys/devices/system/memory/memory%s/state' % block, 'r') as state_file:
+        with open('/sys/devices/system/memory/%s/state' % block, 'r') as state_file:
             if state_file.read() == 'online\n':
                 return True
             return False
 
-    return wait.wait_for(_is_online, timeout=120, step=1) or False
+    return wait.wait_for(_is_online, timeout=2, step=1) or False
```
Comment (ldoktor) [inline avocado/utils/memory.py]: Well this is quite a big change. Are you absolutely sure this is enough for __any__ system under whatever load? I'd suggest at least 10s.

As for the style, please separate this change to a different commit. That improves the readability of the changelog as well as possibility to revert/bisect. (in general when you need to use `also` in commit message, it means the commit is wrong and should be split).

personal note: I like `git cola` or `git citool` for splitting patches, but it's just a personal taste, feel free to use whatever tool that suits you.

File: avocado/utils/memory.py [hunk 3]
```diff
@@ -141,7 +141,7 @@ def memtotal_sys():
     block_size = int(open(os.path.join(sys_mempath,
                                        'block_size_bytes'),
                           "r").read().strip(), 16)
-    return (no_memblocks * block_size)/1024.0
+    return (no_memblocks * block_size) / 1024.0
```
Comment (ldoktor) [inline avocado/utils/memory.py]: Please split those style fixes into a separate commit, it decreases the readability of the change and again breaks reverts as in case of revert this style fix would be gone as well.

File: avocado/utils/memory.py (hunk 2 above)
Comment (abdhaleegit) [reply avocado/utils/memory.py]: well, on my system I see it wont take much time to change the state with load. but it is practical to have 10s wait time.  I am thinking of a way to exit as soon as the state changes to 'online'  instead of waiting 10s even after the state has changed with in 1 second.  so in a case where this function is called for 100 iteration.. 10s x 100 adds up to too much of wait time.  do you suggest some way to exit as early as possible ?

**Comment by abdhaleegit** [comment]
I agree, I will make sure I do not have major changes after the code is out and yes we can have both.

also I will have multiple commits for all the changes

File: avocado/utils/memory.py (hunk 2 above)
Comment (apahim) [reply avocado/utils/memory.py]: You're already using the wait.wait_for, which will interrupt the wait as soon as the `_is_online` returns `True`. With 1s steps, maximum wait period after the status changes to `online` is <1s. The `10s` is just the time we will keep trying, not the total time this call will interrupt the test.

**Comment by abdhaleegit** [comment]
Created a new PR which implements all above comments #2208
